api/user.go

```diff
@@ -954,6 +955,24 @@ func getUser(c *Context, w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+func getUsername(c *Context, w http.ResponseWriter, r *http.Request) {
```

Comment:
What about changing the name of the function to be `getByUsername`? cause we are getting the whole user here and not just the username. make sense?